tensorboard/webapp/header/header_component.ts

```diff
@@ -36,7 +44,9 @@ import {Component} from '@angular/core';
       }
 
       .brand,
-      .settings {
+      settings-button,
+      app-header-reload,
+      a {
```

Comment:
It's potentially confusing to have selectors solely on common tag names.  I would prefer to introduce a class and refer to that instead. On the other hand, if this is a common pattern in the code base and you expect size of each of these components to be small, then potential for future confusion may be limited.

In this case I'm thinking about the "a" selector. but the component-tag selectors seems fine ("settings-button" and "app-header-reload") since you can be pretty assured they will be unique.

Reply:
I am aware of the general principle when writing CSS, but I think that does not apply to Angular and Polymer. The CSS/SCSS is compiled to be scoped by a component so technically, the confusion should almost never happen unless you have more than one instance of a tag.

I can be convinced that we still need semantic names around tags for semantically meaningful selectors.

Reply:
I'm aware that selectors are scoped. It is nonetheless potentially confusing. I'm thinking specifically about the case where someone modifies this component to add another anchor tag. (or modifies the reload_container to add a button or icon).

But if we can assume that these components will be limited in scope or rarely modified then likelihood of confusion decreases. Making changes here is up to you.

Reply:
Done.

Do note, however, that a given component should not be very large in general. When a component becomes unwieldy, one must consider breaking it down to smaller components for readability.